src/Microsoft.IdentityModel.Tokens/TokenValidationParameters.cs

```diff
@@ -455,6 +456,11 @@ public string NameClaimType
         /// </summary>
         public IDictionary<string, object> PropertyBag { get; set; }
 
+        /// <summary>
+        /// Gets or sets the delegate that will be called when reading JSON Web Token header and payload claims.
+        /// </summary>
+        public TryReadJwtClaim TryReadJwtClaim { get; set; }
```

Comment:
The reader will need to know if the runtime is considering the Header or Payload.
Couple of ideas, we can have two delegates one for each, OR we can pass a parameter.

Passing a parameter would work better with SAML tokens that have three statements by default which would lead to three delegates.